packages/react-router/index.ts

```diff
@@ -2,20 +2,34 @@ import * as React from "react";
 import type {
   ActionFunction,
   ActionFunctionArgs,
+  AgnosticRouteObject,
   Blocker,
   BlockerFunction,
+  BrowserHistory,
+  BrowserHistoryOptions,
```

Comment:
We may want to consider exporting these as `UNSAFE_` since the intended usage is `createBrowserRouter` which wraps `createBrowserHistory`/`createRouter`.  Remix currently does the latter for granular control over the hydration/initialization ordering but for the most part histories should be an implementation detail in v7?